Contentstack.Core/ContentstackClient.cs

```diff
@@ -146,21 +159,69 @@ public ContentstackClient(IOptions<ContentstackOptions> options)
         ///     ContentType contentType = stack.ContentType(&quot;contentType_name&quot;);
         /// </code>
         /// </example>
-        public ContentstackClient(string apiKey, string deliveryToken, string environment, string host = null, ContentstackRegion region = ContentstackRegion.US, string version = null) :
+        public ContentstackClient(string apiKey,
+                                string deliveryToken,
+                                string environment,
+                                string host = null,
+                                ContentstackRegion region = ContentstackRegion.US,
+                                string version = null,
+                                string proxyHost = null,
+                                int proxyPort = -1,
+                                ICredentials proxyCredentials = null) :
         this(new OptionsWrapper<ContentstackOptions>(new ContentstackOptions()
         {
             ApiKey = apiKey,
             DeliveryToken = deliveryToken,
             Environment = environment,
             Host = host,
             Region = region,
-            Version = version
+            Version = version,
+            ProxyHost = proxyHost,
+            ProxyPort = proxyPort,
+            ProxyCredentials = proxyCredentials
         }
         ))
         {
 
         }
 
+        protected void Initialize()
+        {
+            HttpClientHandler httpClientHandler = new HttpClientHandler
+            {
+                Proxy = _options.GetWebProxy()
+            };
+            _httpClient = new HttpClient(httpClientHandler);
+
+            _httpClient.DefaultRequestHeaders.Add(HeadersKey.XUserAgentHeader, $"{xUserAgent}");
+
+            if (!_httpClient.DefaultRequestHeaders.UserAgent.Any())
+            {
+                _httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("contentstack-delivery-dotnet", Version));
```

Comment:
Do not copy paste the code. It's not going to work. 

You have to write you own original code. Let's connect for further clarification